nodes/node_diggers.lua

```diff
@@ -225,6 +228,9 @@ minetest.register_node("digtron:intermittent_digger", {
 
 		local facing = minetest.get_node(pos).param2
 		local digpos = digtron.find_new_pos(pos, facing)
+		if nodes_dug:get_pos(digpos) then
+			return 0
+		end
 
 		if protected_nodes:get(digpos.x, digpos.y, digpos.z) then
```

Comment:
`nodes_dug:get_pos` and `protected_nodes:get` are almost always used together. Both check whether `digpos` may be dug, thus I'd propose a helper function to keep it consistent: (untested)
```Lua
local function may_dig_pos(pos, nodes_dug, protected_nodes)
	return not nodes_dug:get_pos(pos) and not protected_nodes:get(pos.x, pos.y, pos.z)
end